Fix stale versioned docs links (#22875)
pingcap/docs: sql-statements/sql-statement-overview.md

> **Note:**
>

> > [TiCDC](https://docs.pingcap.com/tidb/stable/ticdc-overview) & [TiDB Binlog](https://docs-archive.pingcap.com/tidb/v8.3/tidb-binlog-overview/) are tools for replicating TiDB data to the upstream for TiDB Self-Managed. Most SQL statements for TiCDC and TiDB Binlog are not applicable to TiDB Cloud. For TiDB Cloud, you can use the [Changefeed](/tidb-cloud/changefeed-overview.md) feature in the [TiDB Cloud console](https://tidbcloud.com) instead to stream data.

Reviewer: ```suggestion
> [TiCDC](https://docs.pingcap.com/tidb/stable/ticdc-overview) is a tool for replicating TiDB data to the upstream for TiDB Self-Managed. Most SQL statements for TiCDC are not applicable to TiDB Cloud. For TiDB Cloud, you can use the [Changefeed](/tidb-cloud/changefeed-overview.md) feature in the [TiDB Cloud console](https://tidbcloud.com) instead to stream data.
```

Reviewer: reason: "Binlog" related statements had been removed from this table
